### PR TITLE
kinect_fusion_opencl_pose_estimation_pipeline_block: add eigen changes

### PR DESCRIPTION
**Pull request by jordan-heemskerk**
- Adding changes from eigen implementation to opencl version
- Simplify a few lines
- Add force_px_px correspondences option
- Add some tests cases

### CODE REVIEW

**Comment by jordan-heemskerk** [comment]
@RobertLeahy to be reviewed at your leisure

File: cl/pose_estimation.cl [hunk 1]
```diff
@@ -103,20 +103,21 @@ void correspondences_impl(
 	size_t idx=(y*width)+x;
 	idx*=2U;
 
-	float3 curr_pv=vload3(idx,prev_map);
-	if (!is_finite(curr_pv)) {
+	float3 curr_v=vload3(idx,map);
+	float3 curr_n=vload3(idx+1U,map);
```
Comment (RobertLeahy) [inline cl/pose_estimation.cl:107]: This load could be moved lower.

File: src/test/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp [hunk 2]
```diff
@@ -219,7 +268,93 @@ SCENARIO_METHOD(fixture,"dynfu::kinect_fusion_opencl_pose_estimation_pipeline_bl
...
+
+
+			THEN("The kinect_fusion_opencl_pose_estimation_pipeline_block finds the applied perturbation.") {
+
+				auto t_gk=ptr->get(); // what we get back (current cam -> world)
+
+				CHECK((t_gk-gt).isZero());
+
+
+
```
Comment (RobertLeahy) [inline src/test/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp:347]: That's a lot of newlines.

File: src/test/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp [hunk 3]
```diff
@@ -219,7 +268,93 @@ SCENARIO_METHOD(fixture,"dynfu::kinect_fusion_opencl_pose_estimation_pipeline_bl
...
+			}
+
+
+
```
Comment (RobertLeahy) [inline src/test/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp:351]: A lot of newlines here as well.

**Pull request merged**